crates/wasm-mutate-stats/src/bin/wasm-mutate-stats.rs

```diff
@@ -0,0 +1,327 @@
+use itertools::Itertools;
+use md5::{Digest, Md5};
+use rand::{rngs::SmallRng, SeedableRng};
+use serde::{Deserialize, Serialize};
+use std::fmt::Debug;
+use std::fs::File;
+use std::io::prelude::*;
+use std::process::Command;
+use std::{collections::HashMap, fs, sync::Arc};
+use wasm_mutate::WasmMutate;
+struct State {
+    // To print the report
+    print: bool,
+    // To save temporary files like object generation from cranelift
+    save_files: bool,
+    // Tuples of filenames and the corresponding byte stream
+    corpus: Vec<(String, Vec<u8>)>,
+    // Set the arguments passed to `WASMTIME compile <args>`
+    // Each entry of this field corresponds to the <args> above.
+    compiler_configs: Vec<Vec<&'static str>>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+struct ConfigReport {
+    hashes: HashMap<String, Vec<usize>>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+struct Report {
+    // input Wasm file name
+    input: String,
+    // input byte stream
+    wasm: Vec<u8>,
+    // performed rotations
+    rotations: usize,
+    // unique generated wasm hashes and the rotation in which they
+    // were generated
+    hashes: HashMap<String, Vec<usize>>,
+    // extra measurements, one for each compiler configuration
+    configs: Vec<(String, ConfigReport)>,
+    // to save seed and mutation error with that seed
+    errors: Vec<(u64, String)>,
+    // seed of the WasmMutate process
+    seed: u64,
+}
+
+impl Report {
+    /// Adds a new hash to the Wasm hashes map
+    pub fn add_wasm_hash(&mut self, hash: String) {
+        let current = self.rotations;
+        self.hashes
+            .entry(hash)
+            .and_modify(|e| e.push(current))
+            .or_insert(vec![current]);
+    }
+
+    /// Adds a new hash to the specific configuration
+    pub fn add_obj_hash(&mut self, hash: String, idx: usize) {
+        let current = self.rotations;
+        self.configs[idx]
+            .1
+            .hashes
+            .entry(hash)
+            .and_modify(|e| e.push(current))
+            .or_insert(vec![current]);
+    }
+
+    /// Prints the report
+    pub fn print(&self) {
+        println!("Wasm mutate stats for {}", self.input);
+
+        println!(
+            "Unique Wasms {}/{} ({:.2}%)",
+            self.hashes.len(),
+            self.rotations,
+            100 * self.hashes.len() / (self.rotations as usize)
+        );
+
+        for (name, config) in &self.configs {
+            println!(
+                "Unique object files ({}) {}/{} Wasm ({:.2}%)",
+                name,
+                config.hashes.len(),
+                self.hashes.len(),
+                config.hashes.len() as f64 / (self.hashes.len() as f64) * 100.0
+            );
+        }
+    }
+}
+
+const ROTATIONS: usize = 1000000;
+
+fn main() {
+    // Init logs
+    let _ = env_logger::try_init();
```

Comment:
This can be `env_logger::init()` instead of the fallible version, because there is no way anyone else has already initialized a logger, as this is the first line of `main`.